packages/hash/frontend/src/components/GlideGlid/glide-grid.tsx

```diff
@@ -103,6 +119,31 @@ const GlideGrid: ForwardRefRenderFunction<DataEditorRef, DataEditorProps> = (
     [onVisibleRegionChanged],
   );
 
+  const handleColumnResize = useCallback(
+    (column: GridColumn, newSize: number) => {
+      setColumns((prevCols) => {
+        const index = prevCols.findIndex(({ title }) => title === column.title);
+        const newColumns = [...prevCols];
+
+        newColumns.splice(index, 1, {
+          ...prevCols[index]!,
+          width: newSize,
+        });
+        return newColumns;
+      });
```

Comment:
I'm thinking it might just be more sensible to keep the column widths in state (keyed by column id), and then combine them in a `useMemo` at render time. This avoids the need to reset the state when columns change, and also means if you add a new column, the widths of other columns don't reset.